features/field-based-acl.md

```diff
@@ -0,0 +1,190 @@
+Field-based ACL
+================
+
+# API Design Overview
+
+Field-based ACL is a Skygear feature defining the accessibility of the fields
+of a Skygear Record type.
+
+## Accessibility levels
+
+The field-based accessibility is divided into 5 levels:
+
+- `ReadWrite`: fields that available for read and write
+- `Queryable`: Fields that can be used to perform any query predicates and
+  will be returned after query.
+- `Discoverable`: Fields that can only be used to perform equality query
+  predicate and will be returned in query.
```

Comment:
> Also, I think contains operator is not allowed for discoverable-only fields.

You need to allow people to discover user profiles with email like `email IN ["johndoe@example.com", "janedoe@example.com", "johnny.appleseed@example.com"]`.

Interestingly that also means the user can do `gender IN ["male", "female", "indeterminate"]`.

If a field is discoverable, it is possible to enumerate all combinations by making separate requests **if** the number of combinations is small, no matter you support `IN` or not. Therefore, I think we should allow `IN` because the common use case is to discover by email.